src/librustdoc/passes/strip_hidden.rs

```diff
@@ -25,6 +25,28 @@ pub(crate) fn strip_hidden(krate: clean::Crate, cx: &mut DocContext<'_>) -> clea
     let mut retained = ItemIdSet::default();
     let is_json_output = cx.is_json_output();
 
+    let all_items = collect_items(&krate);
+
+    // Collect hidden items
+    let hidden_items: Vec<_> =
+        all_items.iter().filter(|item| item.is_doc_hidden()).cloned().collect();
+
+    // Map import items to their hidden source items
+    let source_items: FxHashMap<_, _> = all_items
```

Comment:
Normally you can use [reexport_chain](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/metadata/struct.ModChild.html#structfield.reexport_chain) to get this info.